pkg/api/v20170131/validate.go

```diff
@@ -100,6 +100,16 @@ func (a *Properties) Validate() error {
 		return e
 	}
 
+	if a.OrchestratorProfile.OrchestratorType == Kubernetes {
+		if a.ServicePrincipalProfile == nil {
+			return fmt.Errorf("ServicePrincipalProfile must be specified with Orchestrator %s", a.OrchestratorProfile.OrchestratorType)
+		}
+
+		if len(a.ServicePrincipalProfile.Secret) == 0 {
+			return fmt.Errorf("service principal client secret must be specified with Orchestrator %s", a.OrchestratorProfile.OrchestratorType)
+		}
+	}
```

Comment:
Also needed in https://github.com/acs-engine/blob/master/pkg/api/v20170701/validate.go?